File: app/src/main/java/com/pr0gramm/app/ui/fragments/FavedCommentFragment.java
package com.pr0gramm.app.ui.fragments;

import com.pr0gramm.app.ActivityComponent;
import com.pr0gramm.app.Settings;
import com.pr0gramm.app.api.pr0gramm.response.Message;
import com.pr0gramm.app.services.FavedCommentService;
import com.pr0gramm.app.services.UserService;
import com.pr0gramm.app.ui.MessageAdapter;
import com.pr0gramm.app.ui.MessageView;

import java.util.List;

import javax.inject.Inject;

import static com.google.common.collect.Lists.transform;

/**
 */
public class FavedCommentFragment extends MessageInboxFragment {
    @Inject
    UserService userService;

    @Inject
    FavedCommentService favedCommentService;

    @Inject
    Settings settings;

    @Override
    protected LoaderHelper<List<Message>> newLoaderHelper() {
        return LoaderHelper.of(() -> {
            return favedCommentService
                    .list(settings.getContentType())
                    .map(comments -> transform(comments, FavedCommentService::commentToMessage));
        });
    }

    @Override
    protected MessageAdapter newMessageAdapter(List<Message> messages) {
        MessageAdapter adapter = super.newMessageAdapter(messages);
        adapter.setPointsVisibility(MessageView.PointsVisibility.NEVER);
        return adapter;
    }

    @Override
    protected void injectComponent(ActivityComponent activityComponent) {
        activityComponent.inject(this);
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/fragments/FavedCommentFragment.kt
package com.pr0gramm.app.ui.fragments

import androidx.lifecycle.lifecycleScope
import com.pr0gramm.app.R
import com.pr0gramm.app.Settings
import com.pr0gramm.app.api.pr0gramm.Message
import com.pr0gramm.app.services.FavedCommentService
import com.pr0gramm.app.services.UserService
import com.pr0gramm.app.ui.MessageAdapter
import com.pr0gramm.app.ui.Pagination
import com.pr0gramm.app.ui.PaginationController
import com.pr0gramm.app.util.di.instance

/**
 */
class FavedCommentFragment : InboxFragment("FavedCommentFragment") {
    private val settings = Settings.get()
    private val favedCommentService: FavedCommentService by instance()
    private val userService: UserService by instance()

    override fun getContentAdapter(): Pair<MessageAdapter, Pagination<Message>> {
        val loader = apiMessageLoader(requireContext()) { olderThan ->
            val username = userService.name
            if (username == null || !userService.isAuthorized) {
                return@apiMessageLoader listOf()
            }

            favedCommentService.list(settings.contentType, username, olderThan).map {
                FavedCommentService.commentToMessage(it)
            }
        }

        // create and initialize the adapter
        val pagination = Pagination(lifecycleScope, loader)

        val adapter = MessageAdapter(
                R.layout.row_inbox_message, actionListener, null,
                PaginationController(pagination))

        return Pair(adapter, pagination)
    }
}
